internal/project/model/oidc_config.go

```diff
@@ -3,6 +3,13 @@ package model
 import (
 	"github.com/caos/zitadel/internal/crypto"
 	es_models "github.com/caos/zitadel/internal/eventstore/models"
+	"strings"
+)
+
+const (
+	http          = "http://"
+	httpLocalhost = "http://localhost"
```

Comment:
to check a local uri: prefix must be one of `http://localhost:` and `http://localhost/`

otherwise your able to register http://localhost-hodor.com